services/image-puller/tests/unit/test_catalog_client.py

```diff
@@ -0,0 +1,51 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+
+import pytest
+from httpx import AsyncClient
+from models_library.docker import DockerImage
+from pytest_httpx import HTTPXMock
+
+from simcore_service_image_puller.catalog_client import (
+    get_images_to_pull,
+    get_shared_client,
+)
+from simcore_service_image_puller.settings import ImagePullerSettings
+
+pytestmark = pytest.mark.asyncio
+
+
+@pytest.fixture
+def catalog_client(settings: ImagePullerSettings) -> AsyncClient:
+    return get_shared_client(settings)
+
+
+@pytest.fixture
+def images_to_pull() -> list[DockerImage]:
+    return ["nginx:latest", "traefik"]
+
+
+@pytest.fixture
+def mock_catalog_client(
+    httpx_mock: HTTPXMock,
+    catalog_client: AsyncClient,
+    images_to_pull: list[DockerImage],
+    hostname: str,
+) -> None:
+    httpx_mock.add_response(
```

Comment:
I thought we used ``respx`` to mock ``httpx`` response (and ``aioreponses`` for ``aiohttp``)
Please let's not add more dependencies if they are not really justified ... you know how complex is the upgrade process ;-)

Reply:
😅 blame it on google, it should that one as the first result. will swap it out, I agree.